backend/src/routes/userRoutes.ts

```diff
@@ -1,8 +1,14 @@
 import express from "express";
-import { getNonce, getUser, updateUser } from "../controllers/userController";
+import {
+  getNonce,
+  getUser,
+  updateUser,
+  searchUser,
+} from "../controllers/userController";
 const userRoutes = express.Router();
 
 userRoutes.get("/nonce/:address", getNonce);
 userRoutes.get("/:address", getUser);
 userRoutes.put("/:address", updateUser);
+userRoutes.get("/search/user", searchUser);
```

Comment:
Here no need to add user at the end. 
the final route will be /api/user/search it's more simple

Reply:
The reason I added it is because the routing somehow gets ambiguous and calls the`userRoutes.get("/:address", getUser);`instead

Reply:
User Routes aren't prefixed by user ?

like /api/user/:address ?

Reply:
it is, but the thing is /api/user/:address and /api/user/search is creating ambiguity for the express router.
/api/user/:address: The :address part is a dynamic segment, meaning it can match any value in that position.
/api/user/search: This is a static path meant to match exactly the word "search".

So the dynamic path takes precedence and proceeds to evaluate the path params as "search"

Reply:
I think the order matter

Reply:
You should put /search before address

Reply:
Ah works! Thanks.